### PR TITLE
ResourceDetectors - Process

### PR DESCRIPTION
**Pull request by Kielek**
Fixes N/A.

## Changes

Initial implementation for `process.pid` detector from https://github.com/open-telemetry/semantic-conventions/blob/main/docs/resource/process.md#process (this attribute is what I need for now).

The plan is to release shortly as alpha with reference to OTel 1.6.0.
Then create follow up `help-wanted` issue to extend it for another attributes and update OTel to 1.7.0.

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* ~~[ ] Design discussion issue #~~
* [x] Changes in public API reviewed

## Notes

Previous initial implementation of resource detector: #1449
You can expect one more similar, for https://github.com/open-telemetry/semantic-conventions/blob/main/docs/resource/host.md, espiecially `host.name`.

### CODE REVIEW

**Review by lachmatt — APPROVED** [review]

File: src/OpenTelemetry.ResourceDetectors.Process/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+`OpenTelemetry.ResourceDetectors.Process` package to be able to use the
+Process Runtime Resource Detectors.
+
+```shell
+dotnet add package OpenTelemetry.ResourceDetectors.Process --prerelease
+```
+
+## Usage
+
+You can configure Process Runtime resource detector to
```
Comment (cijothomas) [inline src/OpenTelemetry.ResourceDetectors.Process/README.md:18]: Please mention that Resource can be applied to all signals, and if feasible show them each here. Or mention that "while Resource can be applied to all signals, this particular snippet is showing how to add for tracing signal".

File: src/OpenTelemetry.ResourceDetectors.Process/ProcessDetector.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+#endif
+using System.Collections.Generic;
+using OpenTelemetry.Resources;
+
+namespace OpenTelemetry.ResourceDetectors.Process;
+
+/// <summary>
+///     Process detector.
+/// </summary>
+public class ProcessDetector : IResourceDetector
```
Comment (cijothomas) [inline src/OpenTelemetry.ResourceDetectors.Process/ProcessDetector.cs:15]:
```suggestion
public sealed class ProcessDetector : IResourceDetector
```

File: src/OpenTelemetry.ResourceDetectors.Process/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Runtime.CompilerServices;
+
+#if SIGNED
+[assembly: InternalsVisibleTo("OpenTelemetry.ResourceDetectors.Process.Tests, PublicKey=002400000480000094000000060200000024000052534131000400000100010051c1562a090fb0c9f391012a32198b5e5d9a60e9b80fa2d7b434c9e5ccb7259bd606e66f9660676afc6692b8cdc6793d190904551d2103b7b22fa636dcbb8208839785ba402ea08fc00c8f1500ccef28bbf599aa64ffb1e1d5dc1bf3420a3777badfe697856e9d52070a50c3ea5821c80bef17ca3acffa28f89dd413f096f898")]
```
Comment (cijothomas) [inline src/OpenTelemetry.ResourceDetectors.Process/AssemblyInfo.cs:7]: nit: which internal field is required by the tests? I think we can avoid that.

Comment (Kielek) [reply src/OpenTelemetry.ResourceDetectors.Process/AssemblyInfo.cs:7]: SemanticConcention is used in tests.
Technically, we can copy values, but it is common practice in this repository. If you think that it is worth to change I will create follow up issue.

File: src/OpenTelemetry.ResourceDetectors.Process/README.md (hunk 1 above)
Comment (Kielek) [reply src/OpenTelemetry.ResourceDetectors.Process/README.md:22]: #1509

File: src/OpenTelemetry.ResourceDetectors.Process/README.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                        // other configurations
+                        .ConfigureResource(resource => resource
+                            .AddDetector(new ProcessDetector()))
+                        .Build();
+```
+
+The resource detectors will record the following metadata based on where
+your application is running:
+
+- **ProcessDetector**: `process.pid`.
```
[the remark this answers is not in the document]
Comment (Kielek) [reply src/OpenTelemetry.ResourceDetectors.Process/README.md:39]: 95dbe3236993155c1e069c2a40df512a442a5758

File: src/OpenTelemetry.ResourceDetectors.Process/AssemblyInfo.cs (hunk 3 above)
Comment (cijothomas) [reply src/OpenTelemetry.ResourceDetectors.Process/AssemblyInfo.cs:7]: This is okay.

File: src/OpenTelemetry.ResourceDetectors.Process/OpenTelemetry.ResourceDetectors.Process.csproj [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <!-- OmniSharp/VS Code requires TargetFrameworks to be in descending order for IntelliSense and analysis. -->
+    <TargetFrameworks>net6.0</TargetFrameworks>
+    <TargetFrameworks Condition="$(OS) == 'Windows_NT'">$(TargetFrameworks);$(NetFrameworkMinimumSupportedVersion)</TargetFrameworks>
+    <Description>OpenTelemetry Extensions - Process Resource Detector.</Description>
+    <MinVerTagPrefix>ResourceDetectors.Process-</MinVerTagPrefix>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="OpenTelemetry" Version="1.6.0" />
```
Comment (cijothomas) [inline src/OpenTelemetry.ResourceDetectors.Process/OpenTelemetry.ResourceDetectors.Process.csproj:10]: `OTelSdkVersion` can be used?

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry.ResourceDetectors.Process/OpenTelemetry.ResourceDetectors.Process.csproj (hunk 5 above)
Comment (Kielek) [reply src/OpenTelemetry.ResourceDetectors.Process/OpenTelemetry.ResourceDetectors.Process.csproj:10]: From PR description:
> The plan is to release shortly as alpha with reference to OTel 1.6.0.
> Then create follow up help-wanted issue to extend it for another attributes and update OTel to 1.7.0.

Will do, just after initial release.

**Pull request merged**